Draft governance model (#3188)
tskit-dev/tskit: GOVERNANCE.md

# Abstract

The purpose of this document is to formalize the governance process used by the
tskit-dev community for core software projects, to clarify who has what responsibilities,
and how decisions are made on changes to those responsibilities.

> tskit-dev is an open and inclusive community. Anyone with an interest in the
> software and science can contribute discussion, and propose changes to code, documentation
> or website content.

Reviewer: ```suggestion
tskit-dev is an open and inclusive community. Anyone with an interest in the
software and science is welcome to contribute discussion, and propose changes to code,
documentation or website content, as long as they follow the code of conduct.
```
I'm adding this to make it sound more inclusive, not less, btw.

# Scope

The governance model described in this document applies to the core
repositories of the tskit software ecosystem. These are:

- [tskit - C and Python API](http://github.com/tskit-dev/tskit)
- [msprime - coalescent simulator](http://github.com/tskit-dev/msprime)
- [tszip - compression library for tskit](http://github.com/tskit-dev/tszip)
- [tskit-site - website for the ecosystem](http://github.com/tskit-dev/tskit-site)
- [kastore - key-value store](http://github.com/tskit-dev/kastore)
- [administrative](http://github.com/tskit-dev/administrative)
- [.github - common config](http://github.com/tskit-dev/.github)

> 

Reviewer 2: Great to have this laid out. Do we want explicitly say the other repos on the tskit-dev workspace (e.g. tscompress, tsinfer) may become the responsibility of other maintainers if need be, or is that too prescriptive?

Author: I thought about this, but decided, as they are out of scope for this document, that it shouldn't say anything about them.

# Roles

## Contributors

> Anyone can become a contributor to the tskit-dev project by, for example,

Reviewer: This sounds like a definition of "contributor" which might be interpreted in various concrete ways (eg having the rights to be on the slack); how about:
```suggestion
Anyone is welcome to contribute to the tskit-dev project by, for example,
```
To be clear, I think the default should be that anyone who contributes should probably be on the slack, but there are situations where they wouldn't be.

- proposing, discussing, or reviewing a change to the code, documentation, or specification
  via a GitHub pull request to the above repositories;

> - reporting a GitHub issue or starting a discussion on the above repositories;

Reviewer 3: What's the process for someone getting added to the github org? (I don't know, just asking the question for visibility)

Author: We need to set this up and have a clean out of permissions across the repos. It's under settings on the org page.

Reviewer 3: I meant what's the process for deciding whether someone should be in the Org or not. That's something we should document

Reviewer: sounds good but in vague terms; too prescriptive might go out of date (as for instance perhaps possible github roles change)

perhaps some general principles - like "the steering committee will choose roles in the github org so that X Y and Z"

Reviewer: ah I see this has been added below

> Potential contributors are encouraged to read and abide by the [CODE OF CONDUCT](https://github.com/tskit-dev/.github/blob/main/CODE_OF_CONDUCT.md).

Reviewer: ```suggestion
Potential contributors are expected to abide by the [CODE OF CONDUCT](https://github.com/tskit-dev/.github/blob/main/CODE_OF_CONDUCT.md).
```

Reviewer: or even "contributors must abide by"

## Maintainers

Maintainers are those who have the "Maintain" role on a given repository. They are able
to merge pull requests, manage issues, and perform other administrative tasks on
that repository. Maintainers are added and removed by a decision of the tskit
Steering Council (TSC). This role is defined on a per-repository basis.

## tskit Steering Council

The tskit Steering Council (TSC) has the following responsibilities:

> - Management of the tskit-dev GitHub organization

Reviewer 3: ```suggestion
- Management of the tskit-dev GitHub organization including the addition of new members
```
